actnow/accounts/permissions.py

```diff
@@ -1,8 +1,20 @@
 from oauth2_provider.models import get_application_model
-from rest_framework.permissions import BasePermission
+from rest_framework.permissions import SAFE_METHODS, BasePermission
 
 
 class AllowAppicationOwnerOnly(BasePermission):
     def has_permission(self, request, view):
         Application = get_application_model()
         return Application.objects.filter(user=request.user.id).exists()
+
+
+class IsOwnerOrReadOnly(BasePermission):
+    def has_object_permission(self, request, view, obj):
+        if request.method in SAFE_METHODS:
+            return bool(request.user.is_active and request.user.is_authenticated)
+        return bool(obj.id == request.user.id)
+
+
+class IsAuthenticated(BasePermission):
```

Comment:
Aren't permissions like this built in?

Reply:
The default `IsAuthenticated` doesn't check if a user is active so this was adding that.

Reply:
Got you. Do inactive users get authenticated? My understanding is if an account is inactive, it shouldn't be able to login i.e. authenticate.

But, if it's possible to authenticate while inactive, then we should either

1. name this permission `IsAuthenticatedAndIsActive`, or
2. Create a standalone `IsActive` permission and then combine it with `IsAuthenticated` when we need both i.e. 
```python
    permission_classes = [IsAuthenticated&IsActive]
```